### ISSUE TITLE
x/crypto/ssh/agent: failure in TestServerResponseTooLarge on plan9-arm builder

### ISSUE THREAD

**Issue by bcmills** [golang/go#50009]
```
-- FAIL: TestServerResponseTooLarge (0.67s)
    client_test.go:347: At least 4 bytes (the response size) should have been successfully written: 0 < 4: write tcp 127.0.0.1:38085->127.0.0.1:42354: write /net/tcp/33/data: i/o on hungup chann
2021/12/03 20:06:22 agent 23: agent: incorrect passphrase
agent.test 2590: warning: process exceeds 100 file descriptors
FAIL
FAIL	golang.org/x/crypto/ssh/agent	6.393s
```

`greplogs --dashboard -md -l -e 'FAIL: TestServerResponseTooLarge'`

[2021-12-03T18:27:20-5770296-c4a8550/plan9-arm](https://build.golang.org/log/09def2993f978244ad1b01383fb7aacd40cef1fe)
<strong>(Note 2-year gap here!)</strong>
[2019-11-16T20:31:45-e1110fd-6ba0be1/plan9-amd64-9front](https://build.golang.org/log/8a6d8ebfb152269011c4e4b95eef65b2451bcb0a)
[2019-03-20T22:39:03-b7391e9-ff04803/plan9-arm](https://build.golang.org/log/78f5554f399d1defbfcbd3f8dd035c8c4fd32a77)
[2019-01-28T20:35:33-c7b33c3-c00595c/plan9-386](https://build.golang.org/log/5e624cc784101cee809927e359d8ec1af13c581e)
[2018-11-10T11:57:17-e4dc69e-6dd70fc/plan9-386](https://build.golang.org/log/ff321b6c659fdf7617d2fa8000997dfaa0f2ec82)

See previously #35888. (CC @fhs @0intro @millerresearch)

**Comment by millerresearch** [comment]
As noted, this seems to be a recurrence of #35888. That was fixed by special case code for Plan 9, which looks for error messages "Hangup" or "i/o on hungup channel". In this new occurrence, it appears the expected message has been truncated to "i/o on hungup chann" -- or is that an artefact of the logging mechanism?
The obvious workaround would be to make the error message comparison more liberal, but it would be good to know what causied the truncation.
I can't replicate this -- perhaps not surprising if it has happened only once in the last 2 years.

**Comment by bcmills** [comment]
Heh, neat! I don't think the truncation here is intentional — this looks like probably a memory-corruption bug. (Maybe a 1-to-0 bit flip somewhere in the path generating the string?)

**Comment by bcmills** [comment]
Duplicate of ##35888

**Comment by millerresearch** [comment]
> probably a memory-corruption bug

There was an issue with syscall error strings being corrupted which should have been fixed by [this commit](https://github.com/golang/go/commit/1b4948). Possibly it's still not quite right, but if so I would have expected more symptoms by now.
